pkg/security/spiffe/spiffe.go

```diff
@@ -24,42 +26,77 @@ import (
 
 // Parsed is a parsed SPIFFE ID according to the Dapr SPIFFE ID path format.
 type Parsed struct {
-	TrustDomain string
-	Namespace   string
-	AppID       string
+	id spiffeid.ID
+
+	namespace string
+	appID     string
 }
 
-// FromContext parses a SPIFFE ID from a gRPC context.
-func FromContext(ctx context.Context) (Parsed, bool, error) {
+// FromGRPCContext parses a SPIFFE ID from a gRPC context.
+func FromGRPCContext(ctx context.Context) (*Parsed, bool, error) {
```

Comment:
Question, why was the Parsed returned value changed to a pointer (heap-allocated)?

Reply:
I thought this would be preferred as it is being passed around a lot.

Reply:
Ok